app/controllers/redirect.php

```diff
@@ -41,13 +49,15 @@ public function perform_action($action, $token)
         */
 
         $customtool = $this->getLtiCustomTool($video, $action);
-        $lti = LtiHelper::getLaunchData($video->config_id, $customtool);
+        $lti = LtiHelper::getLaunchData($video->config_id, $customtool, $video_share);
         if (empty($lti) || empty($customtool)) {
             throw new Error('Es fehlen Parameter!', 422);
         }
 
         // get correct endpoint for redirect type
-        if ($action == 'video') {
+        if ($action == 'share') {
+            $ltilink = self::getLtiLinkFor($lti, 'search');
```

Comment:
The action `video` and `share` are doing the same call to `self::getLtiLinkFor`. Is this correct? If yes, it would make sense to put them int the same if (`if ($action == 'video' || $action == 'share')`)